### DISCUSSION TITLE
Allow Android Backup in Manifest

### DISCUSSION THREAD

**Discussion by Almtesh** [iSoron/uhabits#1580] [Feature Requests]
Hello,

I noticed that the app is not backed up when using the `adb backup` command.  
As far as I know, that's an option you can activate in the Manifest.

**Comment by iSoron** [comment]
Hi @Almtesh, I'm not sure since when, but we already have `android:allowBackup="true"` in our manifest.

**Reply by Almtesh** [reply]
Yes, you're right, I was not watching at the good file.  
`android:allowBackup="true"` is there since 477eb80109804bf6b60176473736916eb95d4fc3, dated November the 8th of 2014.  
Probably the way I'm exploring the backup makes me unable to see it…

**Reply by Almtesh** [reply]
I'm using this bash script to extract my Android backup:
```bash
#!/bin/bash

source=$(realpath "$1")
archive="${source}.tar.gz"
dest="${source}.dir"

if [ -e "${dest}" ];then
	echo "${dest} exists!"
	exit 1
fi
dd if="${source}" ibs=24 skip=1 of="${archive}"
mkdir "${dest}"
cd "${dest}"
printf "\x1f\x8b\x08\x00\x00\x00\x00\x00" | cat - "${archive}" | tar xz
rm "${archive}"
```
